tests/slicebyvalue_test.py

```diff
@@ -247,3 +247,14 @@ def test_label_based_index_to_positional_index_scalar(a):
     a = sc.scalar(a, unit=da.coords['x'].unit)
     ind = label_based_index_to_positional_index(da.sizes, da.coords['x'], a)
     assert_identical(da['x', a], da[ind])
+
+
+def test_label_based_index_to_positional_index_bin_edge_coord_with_scalar():
+    da = sc.data.binned_x(8, 15)
+    assert ('x', 15) == label_based_index_to_positional_index(
+        da.sizes, da.coords['x'], sc.scalar(1, unit='m')
+    )
+    with pytest.raises(IndexError):
+        label_based_index_to_positional_index(
+            da.sizes, da.coords['x'], sc.scalar(-2, unit='m')
+        )
```

Comment:
What about slicing with a range (a `slice` with one or two out if range variables).

Reply:
Also missing a test for values after the last bin.

Reply:
That's already tested in the other tests, specifically `test_label_based_index_to_positional_index`

Reply:
A single value after the last bin should still be tested though.